code/upaya/config/routes.rb

```diff
@@ -1,56 +1,46 @@
-  #   get 'products/:id/purchase' => 'catalog#purchase', as: :purchase
-
-  # Example resource route (maps HTTP verbs to controller actions automatically):
-  #   resources :products
-
-  # Example resource route with options:
-  #   resources :products do
-  #     member do
-  #       get 'short'
-  #       post 'toggle'
-  #     end
-  #
-  #     collection do
-  #       get 'sold'
-  #     end
-  #   end
-
-  # Example resource route with sub-resources:
-  #   resources :products do
-  #     resources :comments, :sales
-  #     resource :seller
-  #   end
-
-  # Example resource route with more complex sub-resources:
-  #   resources :products do
-  #     resources :comments
-  #     resources :sales do
-  #       get 'recent', on: :collection
-  #     end
-  #   end
-
-  # Example resource route with concerns:
-  #   concern :toggleable do
-  #     post 'toggle'
-  #   end
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  # Devise handles login itself. It's first in the chain to avoid a redirect loop during
+  # authentication failure.
+  devise_for :users, skip: [:sessions], controllers: {
+    confirmations: 'users/confirmations',
+    #omniauth_callbacks: 'users/omniauth_callbacks',
+    passwords: 'users/passwords',
+    registrations: 'users/registrations'
+  }
+
+  devise_scope :user do
+    get '/' => 'users/sessions#new', as: :new_user_session
+    post '/' => 'users/sessions#create', as: :user_session
+
+    get 'elis' => 'users/sessions#new'
+    get 'active'  => 'users/sessions#active'
+    get 'timeout' => 'users/sessions#timeout'
+
+    get '/dashboard' => 'dashboard#index', as: :user_root
+
+    patch '/confirm' => 'users/confirmations#confirm'
+
+    get '/users/otp' => 'devise/two_factor_authentication_setup#index'
+    patch '/users/otp' => 'devise/two_factor_authentication_setup#set'
+    get '/users/otp/new' => 'devise/two_factor_authentication#new'
+
+    get '/users/questions' => 'devise/security_questions#new'
+    get '/users/questions/confirm' => 'devise/security_questions#confirm'
+    post '/users/questions/confirm' => 'devise/security_questions#check'
+    match '/users/questions' => 'devise/security_questions#update', via: [:patch, :post]
+
+    get '/users/type' => 'devise/account_type#type'
+    patch '/users/type' => 'devise/account_type#set_type', as: 'set_type'
+    get '/users/type/confirm' => 'devise/account_type#confirm_type'
+  end
+
+  resources :users
+  
+  get '/users/:id/reset_password' => 'users#reset_password', as: :user_reset_password
+
+  if Rails.env.development? || Figaro.env.pt_mode == 'on'
+    mount LetterOpenerWeb::Engine, at: '/letter_opener'
+  end
+
+  root to: 'users/sessions#new'
```

Comment:
I noticed you didn't add any of the SAML routes here, but you've added other SAML-related files in this PR. What do you think about removing them from this PR so that once we add SAML functionality, the PR will include all related files in one place?